models/meshmodel/core/v1alpha1/component_capabilities_db.go

```diff
@@ -0,0 +1,51 @@
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	"github.com/google/uuid"
+)
+
+// This file consists of methods and structs that database(gorm) will use to interact with meshmodel components
+type ComponentDB struct {
+	TypeMeta
+	ComponentSpecDB
+	Metadata []byte `json:"metadata"`
+	// for backward compatibility
+	Spec string `json:"spec,omitempty"`
+}
+
+type ComponentSpecDB struct {
```

Comment:
Please name the structs with the same table names from the object model.

Reply:
Actually there will be two structs. One used to serve client response `ComponentSpec` and another used to create table in the database `ComponentSpecDB`. The reason is that map[string]interface{} is incompatible with sql database. And even if I make the `metadata` field - []byte, the client will not see readable response. This similar approach is already present in some SMP related struct in meshery.

Reply:
Actually this one is a helper struct. `DB` sufficed structs are helper structs which represent the data in the database. Because map[string]interface{} can not be pushed inside sql database and converting it into any other type such as []byte will produce an output unreadable by the client. So the client facing struct is different than this one, This approach is already used in meshery at one other place. It's a little awkward but there is no other way, I could find in my research.